doc/TockBinaryFormat.md

```diff
@@ -0,0 +1,127 @@
+# Tock Binary Format
+
+<!-- npm i -g markdown-toc; markdown-toc -i Compilation.md -->
+
+<!-- toc -->
+
+- [TBF Header](#tbf-header)
+  * [TLV Types](#tlv-types)
+    + [`1` Main](#1-main)
+  * [`2` Writeable Flash Region](#2-writeable-flash-region)
+  * [`3` Package Name](#3-package-name)
+- [Code](#code)
+
+<!-- tocstop -->
+
+Tock processes are represented using the Tock Binary Format (TBF). A TBF
+includes a header portion, which encodes meta-data about the process, followed
+by a binary blob which is executed directly.
+
+## TBF Header 
+
+The TBF header contains a base header, followed by a sequence of
+type-length-value encoded elements. The base header 16 bytes, and has 5 fields:
+
+```
+ 0         2             4            8       12         16
++---------+-------------+------------+-------+----------+
+| Version | Header Size | Total Size | Flags | Checksum |
++---------+-------------+------------+-------+----------+
+```
+
+    * `Version` a 16-bit unsigned integer specifying the TBF header version.
+      Always `2`.
+    * `Header Size` a 16-bit unsigned integer specifying the length of the TBF
+      header in bytes.
+    * `Total Size` a 32-bit unsigned integer specifying the total size of the TBF
+      in bytes (including the header).
+    * `Flags` each bit indicates whether a flag is enabled (1) or disabled (0).
```

Comment:
No guarantee that this will always be only bit flags. Maybe:

```
* `Flags` specifies properties of the process.
 - Bit 0 marks the process enabled. A `1` indicates the process is enabled. Disabled processes will not be launched at startup.
 - Bit 1 marks the process as sticky. A `1` indicates the process is sticky. Sticky processes require the `--force` flag to be erased.
This is useful for services running as processes that should always be available.
- Bits 2-31 are reserved and should be set to 0.
```